### PR TITLE
MAILBOX-386 Solve sync issue in memory event dead letters

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/MemoryEventDeadLetters.java [hunk 1]
```diff
@@ -74,14 +73,14 @@ public MemoryEventDeadLetters() {
         Preconditions.checkArgument(registeredGroup != null, REGISTERED_GROUP_CANNOT_BE_NULL);
 
         synchronized (deadLetters) {
-            return Flux.fromIterable(deadLetters.row(registeredGroup).keySet());
+            return Flux.fromIterable(ImmutableList.copyOf(deadLetters.row(registeredGroup).keySet()));
```
Comment (remk) [inline mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/MemoryEventDeadLetters.java]: You could synchronize only the creation of the immutable list.

Comment (chibenwa) [reply mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/MemoryEventDeadLetters.java]: Not worth the pain of having a local variable/extracted function IMO

Comment (Arsnael) [reply mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/MemoryEventDeadLetters.java]: I agree with @chibenwa on this

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**